### PR TITLE
Add project Sorbet

### PR DESCRIPTION
**Pull request by gdritter-stripe**
This adds the project.yaml as well as the Dockerfile and build.sh to fuzz-test [Sorbet](https://sorbet.org/), an open-source static type-checker for Ruby. Sorbet is an open-sourced project developed by Stripe that's now being used by several Ruby-using companies as well as in open-source projects.

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Can you please provide some details on how Sorbet is being used and customers of it ?

**Comment by gdritter-stripe** [comment]
Absolutely!

Sorbet is a project that originated at Stripe as an effort to build a static type-checker for Ruby code. It is designed to be included in codebases gradually, using a combination of optional annotations along with a file-level annotations to allows programmers to opt-in to various levels of static checking, ranging from `ignore` (at which no static analysis is done at all) up through `strong` (at which point every subexpression in the file must have a known static type), with several intermediate levels.

At this point, Sorbet is used pervasively within Stripe's Ruby codebase: static Sorbet checks are a mandatory part of the CI workflow, and Sorbet also powers editor plugins to do real-time type checking as well as convenience features like autocomplete, as we have leveraged Sorbet's static analysis to power an implementation of the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/). 100% of our production code at Stripe—encompassing millions of lines of code—has some level of static analysis applied to it, and 82% of our production files are annotated as `true` or above in terms of the aforementioned levels, and we are constantly refining the annotations in our codebase in order to catch more errors statically.

Several other organizations—including Shopify, Coinbase, Atrium, the Chan-Zuckerberg Initiative, Gusto, and Kickstarter—are also using Sorbet internally in much the same way. While we only released Sorbet to the public in June, we had previously set up a private beta in order to verify that Sorbet was ready for general use, which means that many of these organizations have been using Sorbet for longer than the public release period, and several of them have also contributed to both Sorbet itself and to other projects build around Sorbet.

There are also several open-source projects that either use or support Sorbet, including [sorbet-rails](https://github.com/chanzuckerberg/sorbet-rails) for building Ruby on Rails applications with Sorbet-powered typechecking, [sord](https://github.com/AaronC81/sord) for turning Yard annotations—informative comments that describe the interfaces of Ruby methods—into Sorbet-compatible annotations, and the shared [sorbet-typed](https://github.com/sorbet/sorbet-typed) repository for adding Sorbet-compatible type annotations to existing Ruby projects which do not include those annotations natively, among others.

Finally, the next major version of Ruby—Ruby 3, which is currently scheduled to be released in December of 2020—is going to feature built-in type annotations to enable this kind of static analysis. The intention is not that Ruby will check static types itself, but rather, that it will standardize a way of writing interfaces which other tools can use. In his [talk at Ruby Kaigi 2019](https://www.youtube.com/watch?v=cmOt9HhszCI), Ruby creator Yukihiro Matsumoto specifically called out Sorbet as such a tool, and we've been collaborating with Matsumoto and the Ruby core team so that Sorbet will be compatible with Ruby 3's standardized notions of typing.

I hope that helps! I'd be happy to expand on any of this or answer any other questions you have!

**Comment by inferno-chromium** [comment]
Sorbet sounds good for integration.

Can you please look into the travis failures for ubsan and afl+asan.

File: projects/sorbet/project.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+homepage: "https://sorbet.org"
+primary_contact: "sorbet@stripe.com"
```
Comment (jonathanmetzman) [inline projects/sorbet/project.yaml]: You should probably add the emails of everyone who will want the crashing testcases to auto_ccs.

**Comment by jonathanmetzman** [comment]
> Please check travis failures for ubsan and afl+asan.

[This line](https://github.com/sorbet/sorbet/blob/a67fb84c40af292048272db8c229e5bb2b8cca08/.bazelrc#L222) seems to be the source of the breakage.

You really should use `$LIB_FUZZING_ENGINE`/`$CFLAGS`/`$CXXFLAGS` and not -`fsanitize=fuzzer,asan` directly in your build files.

If you can't because of the build system (and I don't think this is the first oss-fuzz project to use bazel), can you please leave a comment in the `build.sh` saying sorbet uses `-fsanitize=fuzzer` directly? (You will probably need to disable ubsan and afl builds as well)
It will make it easier for us to know that sorbet is using an unsupported setup if anything goes wrong :-)

**Pull request closed without merging**